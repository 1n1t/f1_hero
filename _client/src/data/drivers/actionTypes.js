/* eslint-disable import/prefer-default-export */
export const UPDATE_DRIVER_LIST = 'drivers/UPDATE_DRIVER_LIST';
