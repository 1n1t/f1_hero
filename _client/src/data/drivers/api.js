/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

export const endPoints = {
	getDriverList: '/api/drivers'
};

export const getDriverList = () => fetchApi(endPoints.getDriverList);
