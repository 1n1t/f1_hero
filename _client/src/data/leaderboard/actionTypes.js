/* eslint-disable import/prefer-default-export */
export const UPDATE_SCORES = 'leaderboard/UPDATE_SCORES';
