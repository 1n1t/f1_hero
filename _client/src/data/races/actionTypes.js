/* eslint-disable import/prefer-default-export */
export const UPDATE_RACE_LIST = 'races/UPDATE_RACE_LIST';
export const SELECT_RACE = 'races/SELECT_RACE';
