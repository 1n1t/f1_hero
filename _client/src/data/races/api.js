/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

export const endPoints = {
	getRaceList: '/api/races'
};

export const getRaceList = () => fetchApi(endPoints.getRaceList);
