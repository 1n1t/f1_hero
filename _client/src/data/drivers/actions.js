/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const updateDriverList = drivers => ({
	type: actionTypes.UPDATE_DRIVER_LIST,
	payload: drivers
});

export const getDriverList = () => async dispatch => {
	const response = await api.getDriverList();
	dispatch(updateDriverList(response.data.drivers));
};
