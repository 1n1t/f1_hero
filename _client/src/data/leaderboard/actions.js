/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

export const updateScores = scores => ({
	type: actionTypes.UPDATE_SCORES,
	payload: scores
});

export const getUsersScores = () => async dispatch => {
	const response = await api.getUsersScores();
	dispatch(updateScores(response.data.leaderboard));
};
