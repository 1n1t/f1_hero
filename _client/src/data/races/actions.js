/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const updateRaceList = races => ({
	type: actionTypes.UPDATE_RACE_LIST,
	payload: races
});

export const selectRace = id => ({
	type: actionTypes.SELECT_RACE,
	payload: id
});

export const updateRace = ({ round, field, value }) => ({
	type: actionTypes.UPDATE_RACE,
	payload: { round, field, value }
});

export const getRaceList = () => async dispatch => {
	const response = await api.getRaceList();
	dispatch(updateRaceList(response.data.races));
};
