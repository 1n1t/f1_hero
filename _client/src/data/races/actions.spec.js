import racesSample from 'JestClient/samples/races';
import t from 'JestClient/actionTester';
import { updateRace, updateRaceList, selectRace, getRaceList } from './actions';
import { UPDATE_RACE_LIST, SELECT_RACE, UPDATE_RACE } from './actionTypes';
import { endPoints } from './api';

const races = racesSample.all;
const { round } = racesSample.selected;

it(`dispatches "${UPDATE_RACE_LIST}" action with correct payload`, () => {
	expect(updateRaceList(races)).toMatchSnapshot();
});

it(`dispatches "${SELECT_RACE}" action with correct payload`, () => {
	expect(selectRace(round)).toMatchSnapshot();
});

it(`dispatches "${UPDATE_RACE}" action with correct payload`, () => {
	const [field, value] = ['alpha3code', 'BAR'];
	expect(updateRace({ round, field, value })).toMatchSnapshot();
});

it('fetches race list from api and call correct action creator', done => {
	const dispatchMock = jest.fn();

	const assertions = (dispatch, request) => {
		expect(request.url).toEqual(endPoints.getRaceList);
		expect(dispatch).toHaveBeenCalledTimes(1);
		expect(dispatch).toHaveBeenCalledWith(updateRaceList(races));
		done();
	};

	t({
		action: getRaceList,
		apiResponse: { races },
		dispatch: dispatchMock,
		assertions
	});
});
