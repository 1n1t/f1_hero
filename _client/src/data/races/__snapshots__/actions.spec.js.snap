// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dispatches "races/SELECT_RACE" action with correct payload 1`] = `
Object {
  "payload": 1,
  "type": "races/SELECT_RACE",
}
`;

exports[`dispatches "races/UPDATE_RACE" action with correct payload 1`] = `
Object {
  "payload": Object {
    "field": "alpha3code",
    "round": 1,
    "value": "BAR",
  },
  "type": "races/UPDATE_RACE",
}
`;

exports[`dispatches "races/UPDATE_RACE_LIST" action with correct payload 1`] = `
Object {
  "payload": Array [
    Object {
      "alpha3code": "AUS",
      "flagUrl": "http://foo.bar",
      "hasPick": true,
      "isPassed": true,
      "round": 1,
      "score": 50,
    },
    Object {
      "alpha3code": "RUS",
      "flagUrl": "http://foo.bar",
      "hasPick": false,
      "isPassed": true,
      "round": 2,
      "score": 0,
    },
    Object {
      "alpha3code": "SPA",
      "flagUrl": "http://foo.bar",
      "hasPick": true,
      "isPassed": false,
      "round": 3,
      "score": 0,
    },
    Object {
      "alpha3code": "BEL",
      "flagUrl": "http://foo.bar",
      "hasPick": false,
      "isPassed": false,
      "round": 4,
      "score": 0,
    },
  ],
  "type": "races/UPDATE_RACE_LIST",
}
`;
